overleaf/clsi
# [ContentCacheManager] use PDF.js Xref table instead of stream detection

### Description

For https://github.com/overleaf/issues/issues/3871

This PR is replacing the `stream`/`endstream` detection with PDF Xref table parsing using PDF.js.

The PDF.js backend was written by @briangough and the final integration was done in pairing session between the two of us.

#### Related Issues / PRs

For https://github.com/overleaf/issues/issues/3871

#### Review

[As discussed in slack](https://digital-science.slack.com/archives/C0216GDEG9L/p1622207748053900): The unit tests are no longer running in a sandbox as it complicates everything.

#### Potential Impact

Low. PDF caching is opt-in / alpha group only.

#### Manual Testing Performed

- small/medium/big project
- see updated unit tests 
- see added snapshot tests for pdfjs logic
